### ISSUE TITLE
[gdx-lwjgl3-angle] Library extraction not compatible with osx packaging

### ISSUE THREAD

**Issue by davidgiga1993** [libgdx/libgdx#7058]
#### Issue details
The angle libraries are extracted in the current directory of the app, see [source code](https://github.com/libgdx/libgdx/blob/999291f7169bf463cee517d911499153cf8f2574/extensions/gdx-lwjgl3-angle/src/com/badlogic/gdx/backends/lwjgl3/angle/ANGLELoader.java#L207)

This causes issues with various package formats since it changes the overall package signature.
On OSX for example, the `*.app` package won't run anymore because of that.

There is no issue when running the plain jar file, but as soon as you're running from a package application everything breaks.

On a more general note it looks like a lot of the native library load code is duplicated from [gdx-jnigen](https://github.com/libgdx/gdx-jnigen/blob/master/gdx-jnigen-loader/src/main/java/com/badlogic/gdx/utils/SharedLibraryLoader.java).

Wouldn't it be possible to simply reuse it instead of re-inventing the wheel?

I noticed the same issue duplicate code issue in the [GlfwAWTLoader](https://github.com/libgdx/libgdx/blob/master/extensions/gdx-lwjgl3-glfw-awt-macos/src/com/badlogic/gdx/backends/lwjgl3/awt/GlfwAWTLoader.java).

Merging the load code would also make sure that the folder/file structure for the native libraries is the same. Currently it's a big mix of everything: Subfolders per os/arch, files per os/arch.

#### Reproduction steps/code
1) Use the angle extension
2) Package the libgdx app into an `*.app` bundle
3) Sign all files
4) Notarize and stamp the bundle
5) Try to start the app

#### Version of libGDX and/or relevant dependencies
1.11.0

#### Stacktrace
```java
EGL: Library not found
```

#### Please select the affected platforms
- [X] macOS

**Comment by badlogic** [comment]
Read the fhll comment. It explaims exactly why this works as it does (which
also explains why it doesn't reuse the loader from jnigen). I don't have a
solution for that, other than forking GLFW, which seems like a worse
solution.

**Comment by davidgiga1993** [comment]
Yeah I read the comment, but the intention isn't completely clear to me. Is it to use the angle glfw binding instead of the regular one? Aka overriding the other library?

The main issue is that it's not usable in this state for osx, since I suppose every dev would ship their app as bundle. Especially due to security requirements by Apple.

**Comment by badlogic** [comment]
Are you integrating the ANGLE binding on purpose? If so, then there
currently is no alternative to how it works.

if you do not use it on purpose, remove it from the Gradle build of your
app.

**Comment by davidgiga1993** [comment]
Okay so I've spent a lot of time trying to find a solution, but it appears that it's not as simple as expected:
1. The default working directory when running an osx app is either `/` or the bundle path itself (depending if executed via finder or `open`)
2. Both location do not allow the libraries to be extracted 

My idea was to disable the extraction at these locations in the `ANGLELoader` and manually include the libs. This however had a couple of other issues since the launcher binary included by `jpackage` doesn't use the `-headerpad_max_install_names` linker flags, it's not possible to use the `install_name_tool` to include the path to the libraries.
This would require a change on the JDK side, which I'll try to push forward.

**Comment by davidgiga1993** [comment]
I was able to successfully include the libraries using an adjusted jdk launcher. I'll write up some documentation on how to include angle in an app bundle. This depends on https://github.com/openjdk/jdk/pull/11922 being merged

**Comment by davidgiga1993** [comment]
The fix was added to the jdk jpackage, so the associated MR here is ready

**Comment by davidgiga1993** [comment]
@crykn can anyone take a look at the code?

**Comment by crykn** [comment]
I don't know anything about the ANGLE backend nor native code, sorry. @badlogic was the one to introduce the gdx-lwjgl3-angle extension, so he's probably the one best equipped to judge #7061.

**Comment by davidgiga1993** [comment]
Hi @badlogic, this MR is still relevant. I'm using the PR in my own fork for quite some time now. The fix in the JDK has been shipped since JDK21 so this MR can be merged.

**Comment by davidgiga1993** [comment]
Just rebased the code and improved the bundle detection to work more reliable
